staging/vhost-device-gpu/src/device.rs

```diff
@@ -1137,61 +1389,240 @@ mod tests {
         backend.set_event_idx(false);
         backend.handle_event(0, EventSet::IN, &[vring], 0).unwrap();
     }
+    }
 
-    #[test]
-    fn test_gpu_command_encode() {
-        let (backend, mem, _) = init();
-        backend.update_memory(mem.clone()).unwrap();
+    mod test_image {
+        use super::*;
+        const GREEN_PIXEL: u32 = 0x00FF00FF;
+        const RED_PIXEL: u32 = 0xFF0000FF;
+        const BYTES_PER_PIXEL: usize = 4;
+
+        pub fn write(mem: &GuestMemoryMmap, image_addr: GuestAddress, width: u32, height: u32) {
+            let mut image_addr: u64 = image_addr.0;
+            for i in 0..width * height {
+                let pixel = if i % 2 == 0 { RED_PIXEL } else { GREEN_PIXEL };
+                let pixel = pixel.as_slice();
```

Comment:
here, I think you should use `to_ne_bytes()` to convert the u32 into a byte array

Reply:
Sure, it is more symmetric with the `assert`. Though technically `as_slice` is from `vm_memory` and it actually uses native byte order.

I'm changing both the `test_image` `assert` and `write` to  `to_be_bytes` since I think this makes the pixel `RED_PIXEL`, and `GREEN_PIXEL` actually red and green :smile:  in the specified `VIRTIO_GPU_FORMAT_R8G8B8A8_UNORM` format. (I always get so confused about which endianess I want :roll_eyes: ) 
Not that it really matters for the test, as long we assert and write in the same order.